src/libambit/device_driver_ambit3.c

```diff
@@ -579,7 +579,7 @@ static int parse_log_header(libambit_sbem0102_data_t *reply_data_object, ambit3_
         data += 19;
         break;
       case 0x8a: /* gen2+gen3 fw */
-        data += 4;
+        data += 1; // Skip size ??
```

Comment:
You're right. We have firmware generation version as a parameter. I'll send you another pull request to fix it and be backward compatible.